### PR TITLE
[FIX] website: add the search params intro the editedObjectPath

### PR DESCRIPTION
**Pull request by jeh-odoo**
In some place the search params are mandatory in the route. We now add the search params to the editedObjectPath to avoid having an error when trying to go back.

task-2993826

### CODE REVIEW

File: addons/website/static/src/systray_items/edit_in_backend.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class EditInBackendSystray extends Component {
 
     editInBackend() {
         const { metadata: { mainObject } } = this.websiteService.currentWebsite;
-        this.websiteService.editedObjectPath = this.websiteService.contentWindow.location.pathname;
+        this.websiteService.editedObjectPath = this.websiteService.contentWindow.location.pathname + this.websiteService.contentWindow.location.search;
```
Comment (rdeodoo) [inline addons/website/static/src/systray_items/edit_in_backend.js:26]: We could also add the hash while at it :) 

```suggestion
        const location = this.websiteService.contentWindow.location;
        this.websiteService.editedObjectPath = location.pathname + location.search + location.hash;
```

**Review by rdeodoo — CHANGES_REQUESTED** [review]
tl;dr: not the correct fix but still something we need :+1: could you apply the remark and we will take care of the underlying issue?
-----------------------

That's not the right fix, this will indeed let you load a page without a crash, but it won't be the expected one.
It will load `/appointment/1/info?xxx` but it should load `/appointment/1` as per the `website_url`.

Investigating a bit your report, there is a bug with that part of the code.
In your case, the `call_button` is correctly returning `{"jsonrpc": "2.0", "id": 31, "result": {"type": "ir.actions.act_url", "url": "/web#action=website.website_preview&path=%2Fappointment%2F1", "target": "self"}}` but somehow the code is not respecting that and decide to instead go to the previously visited URL.

While in your case that underlying issue was a bit hard to spot, since whichever of the 2 url (with or without /info) you land on seems fine, the bug is actually really problematic in some more advanced cases.
Indeed, it did not respect what was asked (the website_url):
- Go to a record in the website
- Click on the edit in backend button
- Navigate from the form view to another model record
- Click on go to website
-> You will land on the last visited URL, so the preview of the first record, not the last one for which you requested the preview.

Concrete case:
- Click on an event and go to register page (the page probably doesn't matter)
- Click on edit in backend
- Click on "3 rooms" stat button
- Click on a room
- On the room form view, click on "Go to website"
- You land on the event page, not the requested room page

But you can ignore that part, your fix will in the meantime avoid the issue, even if not right, as it is still something cool to have

**Review by rdeodoo — APPROVED** [review]
@robodoo r+
Thanks

We will discuss the root cause and fix it here https://github.com/odoo/odoo/pull/101970
